src/dist/replication/lib/replica_file_provider.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017-present, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "replica.h"
+
+#include <fstream>
+
+#include <dsn/dist/block_service.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/filesystem.h>
+
+namespace dsn {
+namespace replication {
+
+// ThreadPool: THREAD_POOL_REPLICATION, THREAD_POOL_REPLICATION_LONG
+error_code replica::do_download(const std::string &remote_dir,
```

Comment:
Why still make do_download in the replica class. Will it be refactored in next PR?

Reply:
`do_download` will also be used in restore, it should not be added into bulk load class, it should be added into a class used to operate remote file provider functions. However, backup and restore is still refactoring. This function is left in replica temporarily.

Reply:
Ok. Let's see how it's implemented, and consider how to refactor it.